collect_app/src/main/java/org/odk/collect/android/widgets/AlignedImageWidget.java

```diff
@@ -103,6 +103,7 @@ protected void setUpLayout() {
         if (splits.length != 6) {
             Timber.w("Only have %d alignment values", splits.length);
         }
+        iarray = new int[6];
```

Comment:
The commit message says only “Fixed bug”. What bug does this fix? And were you able to exercise this code, it being part of AlignedImageWidget, which @lognaturel says requires an external application to run? Perhaps it’s best not to change this file.

Reply:
When it comes to ODK Camera app I tested it and it works like on master branch I can open the app but then there is a crash after capturing an image, so there is something wrong with that app. I doubt someone uses that app.

As you can see in this commit https://github.com/opendatakit/collect/pull/1825/commits/5dedbeeca0d855e37a2e8def4ac311c99ffddb9d I just moved iarray field initialization. Why? because it caused a crash after my refactoring the value of that field was always null and to be honest I have no idea why default initialization was ignored.

Reply:
If we’re not reasonably certain the AlignedImageWidget is never used, then we need to continue to maintain and test it—or at least not change it. Would you agree? Can you share the details of the failure that led you to move the array initialization? If you do keep your change, then you really should make the change I suggested above, otherwise the change causes decay (the code **apparently** showing a naive unawareness that `new int[6]` initializes the array’s elements to 0 [to be clear, I’m not suggesting *you* are either naive or unaware]).

Reply:
You are right when it comes to that `for` loop we could improve that and I did it. I didn't see that because I just fixed the bug I run into. 
Here is the stcktrace:

> java.lang.NullPointerException: Attempt to write to null array
at org.odk.collect.android.widgets.AlignedImageWidget.setUpLayout(AlignedImageWidget.java:110)

that's is the riddle I was talking about on Slack: https://opendatakit.slack.com/archives/C35470318/p1518448269000563

As I sad the default initialization was ignored and it was always null :scream: 

>If we’re not reasonably certain the AlignedImageWidget is never used, then we need to continue to maintain and test it—or at least not change it. Would you agree?

Maybe you are right maybe we should not touch that widget at all since we can't test it well.

To sum: I fixed that small thing with a loop so we can merge it or if you really want I could also revert changes in that `AlignedImageWidget` at all.

Reply:
Can you point me to a point in time in your branch where I can experience the NPE? I’m inclined to get to the bottom of it.

Reply:
Just revert the change I made in this commit and open mentioned `AlignedImageWidget`.

Reply:
Or you can get back to https://github.com/opendatakit/collect/pull/1825/commits/26f6c5751b4be24ba46281f0bdf8197a57860b2d

Reply:
The problem was (or is) that AlignedImageWidget calls super which calls back into AlignedImageWidget’s setUpLayout, which uses a field, iarray, before AlignedImageWidget itself is initialized. 

It might be good to get some other opinions on how best to avoid this. I’m inclined to think that you should avoid manipulating fields before the object is initialized. I can’t recall if and how I’ve dealt with this in the past.

Reply:
Says @ggalmazor:

To solve that I'd separate clearly the construction of the object from its initialization for the UI. During construction, all those fields should receive their value, and the fields should be final wherever it's possible. Then, after creating the instance, we could call `initUI()` or something like that, which would use that field safely

Reply:
initUi should be initUi, says I.